src/SDKs/Azure.Core/data-plane/Azure.Core/Request.cs

```diff
@@ -28,6 +28,14 @@ public virtual void AddHeader(string name, string value)
 
         public abstract bool TryGetHeader(string name, out string value);
 
+        public abstract bool TryGetHeaderValues(string name, out IEnumerable<string> values);
+
+        public abstract bool ContainsHeader(string name);
+
+        public abstract void SetHeader(string name, string value);
+
```

Comment:
Do you see a strong need for having SetHeader as an abstract method on Request?  The implementation you added implements this as a Remove followed by Add.  Could that implementation be provided here as virtual instead of abstract?

Reply:
Yep, virtual sound good.